ui/src/components/helpers/formsHelpers.js

```diff
@@ -113,16 +119,47 @@ const validateForm = async (id, user) => {
         state: { id: data[0].id },
       };
     }
+
+    //Check if user already signed-up for the PAID form
```

Comment:
logic looks fine.

couple improvements i would make:
- move this logic to an API e.g. get-user-submission. Business logic should lie on the server (to prevent business logic attacks)
- i would create a Set instead of an array of emails which would improve the time complexity